pure_protobuf/protobuf.py

```diff
@@ -489,55 +600,83 @@ class TypeMetadataType(Type):
+        )
+        self.__field_metadata_type.add_field(
+            4, "flags", UVarint, flags=Flags.REQUIRED
+        )
+        self.__field_metadata_type.add_field(
+            5, "embedded", EmbeddedMessage(self)
+        )  # Used to describe embedded messages.
         # Metadata message description.
         self.__self_type = EmbeddedMessage(MessageType())
-        self.__self_type.message_type.add_field(1, 'fields', EmbeddedMessage(self.__field_metadata_type), flags=(Flags.REPEATED | Flags.REQUIRED))
-    
+        self.__self_type.message_type.add_field(
+            1,
+            "fields",
+            EmbeddedMessage(self.__field_metadata_type),
+            flags=(Flags.REPEATED | Flags.REQUIRED),
+        )
+
     def __create_message(self, message_type):
-        '''
+        """
         Creates a message that contains info about the message_type.
-        '''
+        """
         message, message.fields = self.__self_type(), list()
         for field in iter(message_type):
             field_meta = self.__field_metadata_type()
-            field_meta.tag, field_meta.name, field_type, field_meta.flags = field
+            field_meta.tag, field_meta.name, field_type, field_meta.flags = (
+                field
+            )
             field_meta.type = type_str = field_type.__class__.__name__
             if isinstance(field_type, EmbeddedMessage):
                 field_meta.flags |= Flags.EMBEDDED
-                field_meta.embedded_metadata = self.__create_message(field_type.message_type)
-            elif not type_str.endswith('Type'):
-                raise TypeError('Type name of type singleton object should end with \'Type\'. Actual: \'%s\'.' % type_str)
+                field_meta.embedded_metadata = self.__create_message(
+                    field_type.message_type
+                )
+            elif not type_str.endswith("Type"):
+                msg = (
+                    "Type name of type singleton object should end with "
+                    "'Type'. Actual: '%s'."
+                ).format(type_str)
+                raise TypeError(msg)
             else:
                 field_meta.type = type_str[:-4]
             message.fields.append(field_meta)
         return message
-    
+
     def dump(self, fp, message_type):
         self.__self_type.dump(fp, self.__create_message(message_type))
-        
+
     def __restore_type(self, message):
-        '''
+        """
         Restores a message type by the information in the message.
-        '''
+        """
         message_type, g = MessageType(), globals()
         for field in message.fields:
-            field_type = field['type']
-            if not field_type in g:
-                raise TypeError('Primitive type \'%s\' not found in this protobuf module.' % field_type)
+            field_type = field["type"]
+            if not isinstance(field_type, str):
+                field_type = field_type.decode("utf-8")
```

Comment:
Perhaps, this was a mine mistake. Name and type should be `Unicode`, not `Bytes`. Then, this decode would be naturally unneeded.